src/test/shell/bazel/remote/remote_execution_tls_test.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+#
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Tests remote execution and caching.
+#
+
+# Load the test setup defined in the parent directory
+CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${CURRENT_DIR}/../../integration_test_setup.sh" \
+  || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
+
+# Create a temporary self-signed certificate
+# Note: 1 cert per test suite is sufficient and more efficient.
+# Note: No need to cleanup as it is created inside a temp test directory.
+cert_path=$(mktemp -d "${TEST_TMPDIR}/sslcert.XXXXXXXX")
+pushd "${cert_path}"
+SERVER_CN=localhost
+CLIENT_CN=localhost
+openssl genrsa -passout pass:1111 -des3 -out ca.key 4096
+openssl req -passin pass:1111 -new -x509 -days 365 -key ca.key -out ca.crt -subj "/CN=${SERVER_CN}"
+openssl genrsa -passout pass:1111 -des3 -out server.key 4096
+openssl req -passin pass:1111 -new -key server.key -out server.csr -subj "/CN=${SERVER_CN}"
+openssl x509 -req -passin pass:1111 -days 365 -in server.csr -CA ca.crt -CAkey ca.key -set_serial 01 -out server.crt
+openssl rsa -passin pass:1111 -in server.key -out server.key
+openssl genrsa -passout pass:1111 -des3 -out client.key 4096
+openssl req -passin pass:1111 -new -key client.key -out client.csr -subj "/CN=${CLIENT_CN}"
+openssl x509 -passin pass:1111 -req -days 365 -in client.csr -CA ca.crt -CAkey ca.key -set_serial 01 -out client.crt
+openssl rsa -passin pass:1111 -in client.key -out client.key
+openssl pkcs8 -topk8 -nocrypt -in client.key -out client.pem
+openssl pkcs8 -topk8 -nocrypt -in server.key -out server.pem
+popd
```

Comment:
Did you copy this code from https://gist.github.com/nddipiazza/64f2c5c447b23038f7cb515cc0af551b?

Two things:
 - We can't just copy code without it having a clear license that we are allowed to use.
 - Instead of generating these every time, generate it once with a 1000 year expiration date and check them into third party with a reference to the commands you used to generate them.

Reply:
Fair point, thanks! Will add a 'permanent' cert

Just for the reference, this is what I used as an exampe: https://github.com/grpc/grpc-java/tree/master/examples/example-tls

Reply:
Ohh nice. In that case we don't have a license issue :-). However, please still generate a cert and check it into third party and add a README to it linking to https://github.com/grpc/grpc-java/tree/master/examples/example-tls

Reply:
done